src/Rest/Routes/LoadMore/AbstractLoadMore.php

```diff
@@ -0,0 +1,253 @@
+		$type = isset($params['type']) ? \sanitize_text_field(\wp_unslash($params['type'])) : '';
+		$initialItems = isset($params['initialItems']) ? \json_decode(\sanitize_text_field(\wp_unslash($params['initialItems']))) : '';
+		$perPageOverride = isset($params['perPageOverride']) ? \sanitize_text_field(\wp_unslash($params['perPageOverride'])) : '';
+		$urlLoadMoreAction = isset($params['urlLoadMoreAction']) ? \sanitize_text_field(\wp_unslash($params['urlLoadMoreAction'])) : 'false';
+		$currentPage = isset($params['currentPage']) ? (int) \sanitize_text_field(\wp_unslash($params['currentPage'])) : 0;
+
+		$response = $this->getLoadMoreData(
+			$query,
+			$type,
+			$initialItems,
+			$urlLoadMoreAction,
+			$currentPage,
+			[
+				'perPageOverride' => $perPageOverride,
+			]
+		);
+
+		if (\is_wp_error($response)) {
+			return \wp_send_json_error(
+				[
+					'code' => $response->get_error_code(),
+					'msg' => $response->get_error_message(),
+				],
+				400
+			);
+		}
+
+		$response['query'] = \wp_json_encode($response['query']);
+
+		return \wp_send_json_success(
+			\array_merge(
+				[
+					'code' => 'success',
+					'msg' => \esc_html__('Success.', 'eightshift-boilerplate'),
+				],
+				$response
+			)
+		);
+	}
+
+	/**
+	 * Get load more data.
+	 *
+	 * @param array<string, mixed> $query WP_Query to request.
+	 * @param string $type Type of usage for later maping.
+	 * @param array<string> $initialItems Initial items that are loaded in dom.
+	 * @param string $urlLoadMoreAction Action to use, on click on on page refresh.
+	 * @param integer $currentPage Current page counter.
+	 * @param array<string, mixed> $config Config array of overides.
+	 *
+	 * @return array<string, mixed>|WP_Error
+	 */
+	private function getLoadMoreData(array $query, string $type, array $initialItems, string $urlLoadMoreAction, int $currentPage, array $config)
+	{
+		// Prepare parameters.
+		$perPageOverride = $config['perPageOverride'] ?? '';
+
+		// Bailout if query is missing.
+		if (!$query) {
+			return new WP_Error(
+				'error_missing_query',
+				\esc_html__('Your request is missing query param. Query param should be a valid WP_Query arguments.', 'eightshift-boilerplate')
+			);
+		}
+
+		// Bailout if type is missing.
+		if (!$type) {
+			return new WP_Error(
+				'error_missing_type',
+				\esc_html__('Your request is missing type param. This is used to filter whate response will the method return.', 'eightshift-boilerplate')
+			);
+		}
+
+		// Bailout if initialItems is missing.
+		if (!$initialItems) {
+			return new WP_Error(
+				'error_missing_initial_items',
+				\esc_html__(
+					'Your request is missing initialItems param. This is used to determin number of items to load and what items to exclude from new requests.',
+					'eightshift-boilerplate'
+				)
+			);
+		}
+
+		// Prepare original query.
+		$originalQuery = $this->prepareOriginalQuery($query, $initialItems, $perPageOverride);
+
+		// Create new query to modify of url load.
```

Comment:
```suggestion
		// Create a new query to modify the load more URL.
```

Is this correct intention of the comment? I'm not sure I understood the original comment.

Reply:
i have updated the comment